src/core/ledger/ledger.js

```diff
@@ -530,8 +549,8 @@ export class Ledger {
     for (const allocation of distribution.allocations) {
       this._allocations.set(allocation.id, allocation);
       this._allocationsToDistributions.set(allocation.id, distribution.id);
-      for (const {id, amount} of allocation.receipts) {
-        this._allocateGrain(id, amount);
+      for (const receipt of allocation.receipts) {
+        this._allocateGrain(receipt, allocation.id, distribution.credTimestamp);
```

Comment:
The distribution's Cred timestamp is the timestamp of the Cred data that was used to distribute the Grain. However, it could be quite far removed from when the Grain was actually distributed. E.g. if we want to turn on SourceCred for a project and retroactively distribute grain for every week in the project's history, we could have a distribution that we make today, but with a cred timestamp from a year ago.

However, having the UI show the Grain as being received a year ago would be confusing-- in this example the Grain is distributed based on last year's cred, but it is distributed today. So I think we should be using the ledger timestamp for the ledger event containing this distribution instead.

Implementing this change will be kind of annoying because the data structure is that the `LedgerEvent` has a timestamp, and the `LedgerEvent` has an `Action`, and this code gets called through a `_processAction` handler that doesn't have the full ledger event. So I think that in order to implement this correctly, we'll need a pre-factor that makes it feasible for action handlers to get access to the full ledger event.

Reply:
Idea: the `_processAction` handler could get the action as the first argument, and the event as an optional second argument. (technically this is redundant because the action is contained within the event, but it would be very convenient in the context of the typed arguments to the action handling functions)